Copy editing: top-level documentation pages

lglattly opened the pull request:
This is the first in what will be a series of pull requests in which I'll go through the full documentation to proofread and copy edit.

We decided to do one pull request per sub-package, so I'll begin on the coordinates package next.

Aside from typos and grammar, I tried to remove any extraneous or hostile language, as well as bring more consistency to formatting such as capitalization in headers or package names.

cc @kelle @adrn

pllim commented:
@bsipocz , if you think this should be backported, feel free to re-milestone. Thanks, @lglattly !

bsipocz commented:
> @bsipocz , if you think this should be backported, feel free to re-milestone.

No, this is tons of changes that are super easy to mess up during backports. I doubt that we'll have than many LTS releases left, and certainly unlikely to have more than 1 on the 3.1.x branch if any.

bsipocz commented on CONTRIBUTING.md:
```diff
@@ -21,56 +21,56 @@ new to this style of development, you'll want to read over our
 Affiliated packages are astronomy-related software packages that are not a part
-of the astropy core package, but build on it for more specialized applications
-and follow the astropy guidelines for reuse, interoperability, and interfacing.
+of the Astropy core package, but build on it for more specialized applications
```
astropy core package is usually refers to astropy the package and usually try to we keep it lowercase/highlighted (whatever else we do in the document with namespaces, class names, etc.)

lglattly replied:
Ah, ok. I didn't realize there were instances where Astropy referred to the core package and was not in backticks. I may need help auditing where it should be used as the proper noun in text and where it should be lowercase (and in backticks?) as the package name.

pllim commented:
You might have to rebase to get rid of the CircleCI failure, which is already fixed on `master`.

bsipocz commented on CONTRIBUTING.md:
```diff
@@ -21,56 +21,56 @@ new to this style of development, you'll want to read over our
 
-- **Code**: the code you are adding, which should follow as much as possible
-  our [coding guidelines](http://docs.astropy.org/en/latest/development/codeguide.html).
+- **Code**: the code you're adding, which should follow
```
why is the you are --> you're changes?

pllim replied:
I have the same question.

bsipocz commented on docs/importing_astropy.rst:
```diff
@@ -1,16 +1,16 @@
-Importing astropy and subpackages
-*********************************
+**********************************
+Importing Astropy and sub-packages
```
keep it lowercase as the package name

adrn commented:
On the latter question: @lglattly and I discussed whether the documentation should be formal text or informal. I suggested that we make it more informal (so those changes are my fault!), but I'd like to hear some discussion about what others think. I don't feel strongly either way, but right now the documentation switches back and forth, so we should probably come up with a stance and stick to it.

kelle commented:
I noticed this too and decided that I like the contractions. It makes it
easier to read (I often read over a “not” but I won’t misread “don’t”) .
And I also like the more casual tone.

My one reservation would be if it makes it harder to read for non-native
English speakers?
-- 
kelle (via iOS)

lglattly commented:
Great points!

Re: astropy vs. Astropy, I wasn't thinking of the package name, just treating it as a proper noun, so that's good to know. I'll fix these instances and differentiate between them moving forward.
Re: you're vs. you are, I'll stick to whatever the general consensus is for formal or informal tone in the documentation.

I'm working on compiling a style guide for contributors as I work through the documentation, so feedback on preferences such as this is very helpful!

lglattly commented:
One other observation related to the formality: I noticed that the top-level docs pages, or more technical pages (like the ones in this PR) tend to have more formal language, whereas the subpackage documentation pages (e.g., coordinates, which has many examples and tutorial-like content) tend to be less formal. So whichever tone we decide on, I'll work on making it more consistent throughout.

bsipocz commented:
I definitely prefer the formal version given that these documents are written mostly for professional astronomers. 
My biggest issue is with "you're" and "it's", it reads super informal, while "don't" is borderline as I often read "do not" as something stronger, but that may be just me.
 
I'm sure git still blames me for some of the discrepancies so either way we consistently go is better than we have now.

pllim commented:
I prefer "do not" because that is what Yoda says, but that is just me.

bsipocz commented:
Checking a few other packages, conclusion 1) they didn't copy edit :) There seem to be a slight preference for the formal, but it's only significant for "it is".
```
$ ~/munka/devel/numpy/doc [hacking|✔] $ grr -i "it is "|wc
     363    4534   37528
$ ~/munka/devel/numpy/doc [hacking|✔] $ grr -i "it's "|wc
      77     932    8118
$ ~/munka/devel/numpy/doc [hacking|✔] $ grr -i "do not "|wc
     100    1212   11087
$ ~/munka/devel/numpy/doc [hacking|✔] $ grr -i "don't "|wc
     119    1423   13619
$ ~/munka/devel/numpy/doc [hacking|✔] $ grr -i "you are "|wc
      55     695    5839
$ ~/munka/devel/numpy/doc [hacking|✔] $ grr -i "you're "|wc
      12     149    1199
```

```
$ ~/munka/devel/scikit-learn/doc [hacking|✔] $ grr -i "it is "|wc
     271    3495   28076
$ ~/munka/devel/scikit-learn/doc [hacking|✔] $ grr -i "it's "|wc
      30     414    3312
$ ~/munka/devel/scikit-learn/doc [hacking|✔] $ grr -i "do not "|wc
      61     769    5941
$ ~/munka/devel/scikit-learn/doc [hacking|✔] $ grr -i "don't "|wc
      36     429    3767
$ ~/munka/devel/scikit-learn/doc [hacking|✔] $ grr -i "you are "|wc
      19     322    2572
$ ~/munka/devel/scikit-learn/doc [hacking|✔] $ grr -i "you're "|wc
      11     135    1245
```

kelle commented:
Having thought about this a bit more and read a little bit, I still think vote contractions and less formal language. I think we want the docs to be welcoming and warm, not cold and formal. 

Sounds like we need more votes so paging @eteq @taldcroft @astrofrog @crawfordsm @mwcraig

bsipocz commented:
I would also ping the other ESL maintainers: @mhvk @saimn @MSeifert04 @nden

mhvk commented:
@lglattly - this is great!  I like the consistency  and even if perhaps I wouldn't have changed "do not" to "don't" either or change the clearly-across-the-ocean "amongst", I think here you get to choose.

astrofrog commented:
Thanks for working on this @lglattly!

I'd like to know what the different levels of formality are, as I'm assuming it isn't a formal/informal binary choice? For instance, I would much rather see the docs use 'you' than speak about users in the third person or use the passive form. On the other hand, I personally find 'you'll' too informal. So for example, my opinion is (using a hypothetical sentence)

*Although astropy is no longer supported on Python 2, users will still be able to install older versions for the foreseeable future using:* (too formal)

*Although astropy is no longer supported on Python 2, you will still be able to install older versions for the foreseeable future using:* (good for me)

*Although astropy is no longer supported on Python 2, you'll still be able to install older versions for the foreseeable future using:* (too informal)

My personal vote is informal enough that we address users directly with 'you', but minimizing the use of contractions, especially e.g. *you'll* or *you're*. But I guess my main question here is whether there are well defined sub-levels in informal and formal style that we can pick from?

taldcroft commented:
I'm definitely :-1: on doing all the contractions.  I think it is easier for non-English speakers to read and more straightforward.  As a documentation writer I would be unhappy to be forced by style rules to write that way.  So yes, I prefer a more formal tone for docs, like the second example from @astrofrog.

nden commented:
👍 for the second example @astrofrog gave. For me this is the right level of formality. As a non-native English speaker I find it a bit easier to read without contractions. For one, the tense becomes obvious, for example: _she's gone_: _she is gone_ or _she has gone_

mhvk commented:
FWIW, I don't have a strong preference, but python itself doesn't seem to do contractions, at least not in their tutorial: https://docs.python.org/3/tutorial/index.html (which has "you will" twice).

pllim commented:
American = less typing, so it gets my vote!

lglattly commented:
> Thanks for working on this @lglattly!
> 
> I'd like to know what the different levels of formality are, as I'm assuming it isn't a formal/informal binary choice? For instance, I would much rather see the docs use 'you' than speak about users in the third person or use the passive form. On the other hand, I personally find 'you'll' too informal. So for example, my opinion is (using a hypothetical sentence)
> 
> _Although astropy is no longer supported on Python 2, users will still be able to install older versions for the foreseeable future using:_ (too formal)
> 
> _Although astropy is no longer supported on Python 2, you will still be able to install older versions for the foreseeable future using:_ (good for me)
> 
> _Although astropy is no longer supported on Python 2, you'll still be able to install older versions for the foreseeable future using:_ (too informal)
> 
> My personal vote is informal enough that we address users directly with 'you', but minimizing the use of contractions, especially e.g. _you'll_ or _you're_. But I guess my main question here is whether there are well defined sub-levels in informal and formal style that we can pick from?
> 
> It might be worth checking whether there are already existing style guidelines for other projects e.g. Python itself or other 'big' packages?

Great examples! Unfortunately there are no clearcut official levels of formality -- it's up to each organization to develop their own style guide for their documents and writing based on the two major style guides (Associated Press and Chicago Manual of Style) and their own personal preferences.

In looking into the matter a little more, I've found that AP recommends using contractions in any kind of writing that's meant to be more conversational, such as journalistic writings and blogs, whereas in professional writing like legal contracts and academic papers, contractions should be avoided. CMS doesn't pass judgement on the use of contractions, but suggests that writers use them when they make writing sound more natural. But all the blogs/articles I've looked at seem to agree that in legal documents and instances where the audience includes non-native English speakers, contractions should be avoided.

So it sounds like we need to decide whether the documentation tone should match the tone of the other materials for consistency across Astropy or if the documentation should be set apart as more formal. I see consistency as the main pro for using contractions in the documentation, especially since there is tutorial-like material in the documentation and contractions make the reader feel like you are talking to them directly, which is beneficial for learning materials. But the main con with using contractions is making it more difficult for non-native English speakers.

And if we decide we want to develop two different style guides for contributors, one for documentation and one for other materials, that's also an option!

bsipocz commented:
> contractions make the reader feel like you are talking to them directly

fyi, it never made me feel it's talking to me directly, I suppose partly it becomes the game of grammar guessing (Nadia mentioned before) and it takes mental energy away from the docs, and going too informal (e.g. many places in openastronomy pages) hugely annoys me.

astrofrog commented:
> But the main con with using contractions is making it more difficult for non-native English speakers.

In my view, supporting non-native English speakers is very important for the project in terms of international inclusiveness (we could lose users who find the docs too hard to read, but we won't likely users who don't find the docs not conversational enough), so given this I vote for not having contractions. If we do use them in other materials (tutorials etc) then we should perhaps consider changing these too. The core docs seems like a good place to start setting precedent for style.

adrn commented:
I slightly prefer contractions, but in the interest of being inclusive toward non-native speakers, it sounds like "no contractions" is the way to go. Thanks everyone!

lglattly commented:
I went through and addressed the feedback on this pull request with second round edits -- I'm tracking the editorial decisions and style preferences in a style guide I'm compiling, that I'll be able to share for feedback soon!

adrn commented on docs/importing_astropy.rst:
```diff
@@ -1,16 +1,16 @@
-*********************************
+**********************************
+Importing `astropy` and sub-packages
+**********************************
```
Here, the "***" have to extend as far as the text (the circleCI test failure is because of this)

adrn commented on docs/install.rst:
```diff
@@ -92,21 +94,22 @@ And the following packages can optionally be used when testing:
+installed.
 
-Installing Astropy
+Installing `astropy`
```
Here the underline has to extend to the end of the text as well

adrn commented on docs/install.rst:
```diff
@@ -115,52 +118,52 @@ can also do::
 
-Testing an installed Astropy
+Testing an Installed `astropy`
 ----------------------------
```
Extend underline

adrn commented on docs/install.rst:
```diff
@@ -323,64 +326,64 @@ The C libraries currently bundled with Astropy include:
 
-Installing Astropy into CASA
+Installing `astropy` into CASA
 ----------------------------
```
Extend underline

adrn commented on docs/known_issues.rst:
```diff
@@ -266,11 +267,11 @@ If so, you can go ahead and try running ``setup.py`` again (in the new
 
-Creating a Time object fails with ValueError after upgrading Astropy
+Creating a Time Object Fails with ValueError After Upgrading `astropy`
 --------------------------------------------------------------------
```
Extend underline

adrn commented on docs/importing_astropy.rst:
```diff
@@ -1,16 +1,16 @@
-Importing astropy and subpackages
-*********************************
+************************************
+Importing `astropy` and Sub-packages
```
When astropy (lowercase) appears in a title like this, I think you have to do 2 backticks: 
```suggestion
Importing ``astropy`` and Sub-packages
```

adrn commented on docs/install.rst:
```diff
@@ -92,21 +94,22 @@ And the following packages can optionally be used when testing:
 
-Installing Astropy
-==================
+Installing `astropy`
```
```suggestion
Installing ``astropy``
```

adrn commented on docs/install.rst:
```diff
@@ -92,21 +94,22 @@ And the following packages can optionally be used when testing:
-Installing Astropy
-==================
+Installing `astropy`
+====================
```
```suggestion
======================
```

adrn commented on docs/importing_astropy.rst:
```diff
@@ -1,16 +1,16 @@
-*********************************
+************************************
+Importing `astropy` and Sub-packages
+************************************
```
```suggestion
**************************************
```

adrn commented on docs/importing_astropy.rst:
```diff
@@ -1,16 +1,16 @@
-*********************************
-Importing astropy and subpackages
-*********************************
+************************************
```
```suggestion
**************************************
```

adrn commented on docs/install.rst:
```diff
@@ -115,52 +118,52 @@ can also do::
 
-Testing an installed Astropy
-----------------------------
+Testing an Installed `astropy`
```
```suggestion
Testing an Installed ``astropy``
```

adrn commented on docs/install.rst:
```diff
@@ -115,52 +118,52 @@ can also do::
-Testing an installed Astropy
-----------------------------
+Testing an Installed `astropy`
+------------------------------
```
```suggestion
--------------------------------
```

adrn commented on docs/install.rst:
```diff
@@ -323,64 +326,64 @@ The C libraries currently bundled with Astropy include:
 
-Installing Astropy into CASA
-----------------------------
+Installing `astropy` into CASA
```
```suggestion
Installing ``astropy`` into CASA
```

adrn commented on docs/install.rst:
```diff
@@ -323,64 +326,64 @@ The C libraries currently bundled with Astropy include:
-Installing Astropy into CASA
-----------------------------
+Installing `astropy` into CASA
+-------------------------------
```
```suggestion
---------------------------------
```

adrn commented on docs/install.rst:
```diff
@@ -461,7 +464,7 @@ can also open issues in the repositories for some of the dependencies:
 .. _sourcebuildtest:
 
-Testing a source code build of Astropy
+Testing a Source Code Build of Astropy
```
```suggestion
Testing a Source Code Build of ``astropy``
```

adrn commented on docs/install.rst:
```diff
@@ -461,7 +464,7 @@ can also open issues in the repositories for some of the dependencies:
 
-Testing a source code build of Astropy
+Testing a Source Code Build of Astropy
 --------------------------------------
```
```suggestion
------------------------------------------
```

adrn commented on docs/known_issues.rst:
```diff
@@ -266,11 +267,11 @@ If so, you can go ahead and try running ``setup.py`` again (in the new
 
-Creating a Time object fails with ValueError after upgrading Astropy
---------------------------------------------------------------------
+Creating a Time Object Fails with ValueError After Upgrading `astropy`
```
```suggestion
Creating a Time Object Fails with ValueError After Upgrading ``astropy``
```

adrn commented on docs/known_issues.rst:
```diff
@@ -266,11 +267,11 @@ If so, you can go ahead and try running ``setup.py`` again (in the new
-Creating a Time object fails with ValueError after upgrading Astropy
---------------------------------------------------------------------
+Creating a Time Object Fails with ValueError After Upgrading `astropy`
+----------------------------------------------------------------------
```
```suggestion
------------------------------------------------------------------------
```

bsipocz commented:
Would be nice to squash the latest commits added from the review. Otherwise the review suggestions can be accepted into one single commit if it's done in the "files changed" view rather than in the main "conversation" part of the PR.

bsipocz commented on docs/importing_astropy.rst:
```diff
@@ -1,16 +1,16 @@
 
 Since most of the functionality in Astropy resides in sub-packages, importing
-astropy as::
+`astropy` as::
```
```suggestion
``astropy`` as::
```

I'm not exactly sure why it triggers a warning from this file, but not from the other one.

adrn commented:
I squashed those accepted suggestion commits.

Also, @lglattly, I was wrong about the single backticks around `astropy` - they should actually be double backticks, so I added some commits to deal with changing those back - sorry for the confusion!

adrn commented:
In the words of @lglattly: "So, do pull requests ever end?!?" 😄 - congrats on your first contribution to astropy core and thanks for your work on this! On to the rest of the documentation :)

bsipocz commented:
> @lglattly: "So, do pull requests ever end?!?"

watch out for the frantic merging next week for feature freeze, and the weeks following it with last minute bug fixes :)


But seriously, thank you for taking care of the docs!
